### PR TITLE
fix: update links when perform "Make Doc Folder Note" and "Link to Parent Folder"

### PR DESCRIPTION
**Pull request by anselmwang**
fix issue #13 : "Make Doc Folder Note" and "Link to Parent Folder" does not update links.

Does this fix look good to you?

### CODE REVIEW

File: src/modules/resolver.ts [hunk 1]
```diff
@@ -20,6 +20,14 @@ export default class NoteResolver {
     return this.plugin.app.vault;
   }
 
+  private rename(file: TFile, newNotePath: string)
+  {
+    let old_file = this.vault.getAbstractFileByPath(file.path);
```
Comment (aidenlx) [inline src/modules/resolver.ts]: this step should return the same TFile instance as the input `file`

File: src/modules/resolver.ts [hunk 2]
```diff
@@ -165,7 +173,7 @@ export default class NoteResolver {
         shouldRun = fnPath && !this.getFolderNote(file.parent);
       if (shouldRun && !dryrun) {
         const { path } = fnPath;
-        this.vault.rename(file, path);
+        this.rename(file, path);
```
Comment (aidenlx) [inline src/modules/resolver.ts]: maybe try replacing `this.vault.rename` with `this.plugin.app.fileManager.renameFile` directly?

File: src/modules/resolver.ts [hunk 3]
```diff
@@ -242,7 +250,9 @@ export default class NoteResolver {
         default:
           assertNever(this.settings.folderNotePref);
       }
-      if (newNotePath) this.vault.rename(file, newNotePath);
+      if (newNotePath) {
+        this.rename(file, newNotePath);
```
Comment (aidenlx) [inline src/modules/resolver.ts]: same as above

**Review by aidenlx — COMMENTED** [review]
simplify code (see files changed for details)

File: src/modules/resolver.ts (hunk 1 above)
Comment (anselmwang) [reply src/modules/resolver.ts]: resolved

**Comment by anselmwang** [comment]
Updated. Just realize `TFile` can be casted to `TAbstractFile`. 

Does it look good now?

**Comment by aidenlx** [comment]
Everything else look good to me, I’ll push a new version about this fix after the merge. Thanks in advance!

**Pull request merged**